tests/robustness/traffic/client.go

```diff
@@ -66,49 +76,62 @@ func (c *RecordingClient) Get(ctx context.Context, key string) (*mvccpb.KeyValue
 }
 
 func (c *RecordingClient) Range(ctx context.Context, key string, withPrefix bool) ([]*mvccpb.KeyValue, error) {
-	callTime := time.Since(c.baseTime)
 	ops := []clientv3.OpOption{}
 	if withPrefix {
 		ops = append(ops, clientv3.WithPrefix())
 	}
+	c.mux.Lock()
+	callTime := time.Since(c.baseTime)
 	resp, err := c.client.Get(ctx, key, ops...)
-	returnTime := time.Since(c.baseTime)
 	if err != nil {
+		c.mux.Unlock()
 		return nil, err
 	}
+	returnTime := time.Since(c.baseTime)
 	c.history.AppendRange(key, withPrefix, callTime, returnTime, resp)
+	c.mux.Unlock()
 	return resp.Kvs, nil
 }
 
 func (c *RecordingClient) Put(ctx context.Context, key, value string) error {
+	c.mux.Lock()
```

Comment:
minor comment: defer unlock?  This comment applies the following couple of places